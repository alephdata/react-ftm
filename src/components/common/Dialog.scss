.Dialog {
  padding-bottom: 0 !important;

  &.large {
    width: 800px;
  }

  &__content {
    position: relative;
    padding-bottom: 20px;

    &.isProcessing {
      pointer-events: none;
    }
  }

  &__loading-overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: rgba(white, 0.7);
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;

    .bp4-spinner {
      top: 50%;
      position: relative;
      transform: translateY(-50%);
    }
  }
}
