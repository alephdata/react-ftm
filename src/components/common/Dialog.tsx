import * as React from 'react'
import { Dialog as Bp3Dialog, IconName, Spinner } from '@blueprintjs/core'
import c from 'classnames';

import "./Dialog.scss";

interface IDialogProps {
  icon: IconName,
  isOpen: boolean,
  isProcessing: boolean,
  title: string,
  onClose: () => any,
  className?: string,
}

export class Dialog extends React.Component<IDialogProps> {
  render() {
    const { children, className, isProcessing, ...rest } = this.props;

    return (
      <Bp3Dialog
        className={c('Dialog', className)}
        portalClassName="dialog-portal-container"
        {...rest}
      >
        <div className={c('Dialog__content', isProcessing)}>
          {children}
          {isProcessing && (
            <div className="Dialog__loading-overlay">
              <Spinner />
            </div>
          )}
        </div>
      </Bp3Dialog>
    );
  }
}
