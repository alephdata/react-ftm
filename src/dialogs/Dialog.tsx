import * as React from 'react'
import { Dialog as Bp3Dialog, IconName, Spinner } from '@blueprintjs/core'
import { Schema } from '@alephdata/followthemoney'
import c from 'classnames';

import "./Dialog.scss";

interface IDialogProps {
  icon: IconName,
  isOpen: boolean,
  isProcessing: boolean,
  title: string,
  toggleDialog: () => any,
}

class Dialog extends React.Component<IDialogProps> {
  render() {
    const { children, icon, isOpen, isProcessing, title, toggleDialog } = this.props;

    return (
      <Bp3Dialog
        icon={icon}
        isOpen={isOpen}
        title={title}
        onClose={toggleDialog}
        className="Dialog"
        portalClassName="dialog-portal-container"
      >
        <div className={c('Dialog__content', isProcessing)}>
          {children}
          {isProcessing && (
            <div className="Dialog__loading-overlay">
              <Spinner />
            </div>
          )}
        </div>
      </Bp3Dialog>
    );
  }
}

export default Dialog;
