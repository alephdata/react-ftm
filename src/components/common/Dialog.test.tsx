import React from 'react';
import { render, screen } from '@testing-library/react';
import { Dialog } from './Dialog';

const renderDialog = (props: object = {}) => {
  return render(
    <Dialog
      icon="info-sign"
      title="Lorem Ipsum"
      isProcessing={false}
      isOpen={true}
      onClose={() => {
        /* no-op */
      }}
      {...props}
    >
      Hello World
    </Dialog>
  );
};

describe('<Dialog />', () => {
  it('renders correctly with no properties', () => {
    renderDialog();
    expect(screen.getByText('Hello World')).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });

  it('renders custom classes', () => {
    renderDialog({ className: 'my-dialog' });
    expect(document.body).toMatchSnapshot();
    expect(document.querySelectorAll('.my-dialog')).toHaveLength(1);
  });
});
