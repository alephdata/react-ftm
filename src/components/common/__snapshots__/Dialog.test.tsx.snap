// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> renders correctly with no properties 1`] = `
<body
  class="bp3-overlay-open"
>
  <div />
  <div
    class="bp3-portal dialog-portal-container"
  >
    <div
      aria-live="polite"
      class="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
    >
      <div
        class="bp3-overlay-start-focus-trap bp3-overlay-appear bp3-overlay-appear-active"
        tabindex="0"
      />
      <div
        class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
      />
      <div
        class="bp3-dialog-container bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
      >
        <div
          aria-labelledby="title-bp-dialog-0"
          class="bp3-dialog Dialog"
          role="dialog"
        >
          <div
            class="bp3-dialog-header"
          >
            <span
              aria-hidden="true"
              class="bp3-icon bp3-icon-info-sign"
              icon="info-sign"
              tabindex="-1"
            >
              <svg
                data-icon="info-sign"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zM9 4h2v2H9V4zm4 12H7v-1h2V8H8V7h3v8h2v1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <h4
              class="bp3-heading"
              id="title-bp-dialog-0"
            >
              Lorem Ipsum
            </h4>
            <button
              aria-label="Close"
              class="bp3-button bp3-minimal bp3-dialog-close-button"
              type="button"
            >
              <span
                aria-hidden="true"
                class="bp3-icon bp3-icon-small-cross"
                icon="small-cross"
              >
                <svg
                  data-icon="small-cross"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="Dialog__content"
          >
            Hello World
          </div>
        </div>
      </div>
      <div
        class="bp3-overlay-end-focus-trap bp3-overlay-appear bp3-overlay-appear-active"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;

exports[`<Dialog /> renders custom classes 1`] = `
<body
  class="bp3-overlay-open"
>
  <div />
  <div
    class="bp3-portal dialog-portal-container"
  >
    <div
      aria-live="polite"
      class="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
    >
      <div
        class="bp3-overlay-start-focus-trap bp3-overlay-appear bp3-overlay-appear-active"
        tabindex="0"
      />
      <div
        class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
      />
      <div
        class="bp3-dialog-container bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
      >
        <div
          aria-labelledby="title-bp-dialog-1"
          class="bp3-dialog Dialog my-dialog"
          role="dialog"
        >
          <div
            class="bp3-dialog-header"
          >
            <span
              aria-hidden="true"
              class="bp3-icon bp3-icon-info-sign"
              icon="info-sign"
              tabindex="-1"
            >
              <svg
                data-icon="info-sign"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zM9 4h2v2H9V4zm4 12H7v-1h2V8H8V7h3v8h2v1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <h4
              class="bp3-heading"
              id="title-bp-dialog-1"
            >
              Lorem Ipsum
            </h4>
            <button
              aria-label="Close"
              class="bp3-button bp3-minimal bp3-dialog-close-button"
              type="button"
            >
              <span
                aria-hidden="true"
                class="bp3-icon bp3-icon-small-cross"
                icon="small-cross"
              >
                <svg
                  data-icon="small-cross"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="Dialog__content"
          >
            Hello World
          </div>
        </div>
      </div>
      <div
        class="bp3-overlay-end-focus-trap bp3-overlay-appear bp3-overlay-appear-active"
        tabindex="0"
      />
    </div>
  </div>
</body>
`;
